kubernetes/nodecollection.go

```diff
@@ -86,37 +87,94 @@ func (cns ClientsetNodeSource) NodeAddress(node *v1.Node) (string, int32, error)
 	return "", 0, fmt.Errorf("Could not find internal IP address for node %s ", node.Name)
 }
 
-// Endpoint an enumeration representing the various metrics endpoints
-type Endpoint uint8
+// Connection is a bitmask that describes the manner(s) in which
+// the agent can connect to an endpoint
+type Connection uint8
+
+const (
+	Unreachable Connection = 0
```

Comment:
Good use of the default value. This might be better at the end of the `const` block as this will make the first mask value `1` rather than `2`.